deephaven/vscode-deephaven
# feat: Connect to DHE worker

Initial DHE connection feature:
- Connect to configured DHE server
- Prompt for credentials
- Create / delete core+ workers (currently using default settings)
- Determine worker type to create based on active editor languageid
- Auto dispose on vscode restart

### Testing
Setup
- Connect to VPN, and have an available DHE server (dev-sanluis is what I've been using)
  > Note It can be helpful to open a Query Monitor on this server to see workers get created / destroyed via the extension in the steps that follow.
- Configure the DHE server in vscode user or workspace settings:
  ```jsonc
  {
    "deephaven.enterpriseServers": [
      "https://dev-sanluis.int.illumon.com:8123/"
    ],
  }
  ```
- Pull this branch, npm install packages, then f5 for debugging. It should automatically open another vscode instance. In that instance, open a vscode folder containing .py / .groovy scripts, or create a new one where files can be put (vscode should remember this folder the next time you run the debugger)

- Click the DH activity bar item. You should see your configured server in the list of "Running" servers:
  <img width="476" alt="image" src="https://github.com/user-attachments/assets/7ccae62f-4818-496a-9d32-bd722be99e51">

- Click on the server to initiate login flow. Provide credentials
- A worker should be created (should see it loading in the CONNECTIONS panel
  ![image](https://github.com/user-attachments/assets/d51bb799-3625-41ca-91b2-e0083493483f)
- Create a new python file with a simple script:
  ```python
  from deephaven import time_table

  simple_ticking = time_table("PT2S")
  ```
  
- Run it using double arrows at top of file.
- If you don't have any other configured, running server, it should automatically connect to the worker you started. Otherwise, you will be prompted to select your connection  
  <img width="723" alt="image" src="https://github.com/user-attachments/assets/f5054c0d-8451-4915-8612-94845948d37c">
- To discard the worker, you cal use the trash icon in the connection tree item
  
  <img width="426" alt="image" src="https://github.com/user-attachments/assets/eb34a7e9-231a-4019-b6be-4a6759b74d62">
